dotnet/src/Connectors/Connectors.Gemini/TextGeneration/GeminiTextGenerationService.cs

```diff
@@ -0,0 +1,134 @@
+/// </summary>
+public sealed class GeminiTextGenerationService : ITextGenerationService
+{
+    private readonly Dictionary<string, object?> _attributes = new();
+    private readonly string _model;
+    private readonly HttpClient _httpClient;
+    private readonly string _apiKey;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GeminiTextGenerationService"/> class.
+    /// </summary>
+    /// <param name="model">The model identifier.</param>
+    /// <param name="apiKey">The API key.</param>
+    /// <param name="httpClient">The optional HTTP client.</param>
+    public GeminiTextGenerationService(string model, string apiKey, HttpClient? httpClient = null)
+    {
+        Verify.NotNullOrWhiteSpace(model);
+        Verify.NotNullOrWhiteSpace(apiKey);
+
+        this._model = model;
+        this._apiKey = apiKey;
+        this._httpClient = HttpClientProvider.GetHttpClient(httpClient);
+        this._attributes.Add(AIServiceExtensions.ModelIdKey, this._model);
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyDictionary<string, object?> Attributes => this._attributes;
+
+    /// <inheritdoc />
+    public Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        return this.InternalGetTextContentsAsync(prompt, executionSettings, cancellationToken);
+    }
+
+    private async Task<IReadOnlyList<TextContent>> InternalGetTextContentsAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken cancellationToken)
+    {
+        using var httpRequestMessage = this.VerifyArgumentsAndGetHTTPRequestMessage(prompt, executionSettings);
+
+        using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        var body = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+
+        var geminiResponse = JsonSerializer.Deserialize<GeminiResponse>(body);
+        if (geminiResponse is null)
+        {
+            throw new KernelException("Unexpected response from model")
+            {
+                Data = { { "ResponseData", body } },
+            };
+        }
+
+        return geminiResponse.Candidates.Select(c => new TextContent(c.Content.Parts[0].Text,
+            this.GetModelId(), geminiResponse, metadata: GetResponseMetadata(geminiResponse, c))).ToList();
+    }
+
+    private static Dictionary<string, object?> GetResponseMetadata(
+        GeminiResponse geminiResponse,
+        GeminiResponseCandidate candidate) => new()
+    {
+        ["FinishReason"] = candidate.FinishReason,
+        ["Index"] = candidate.Index,
+        ["TokenCount"] = candidate.TokenCount,
+        ["SafetyRatings"] = candidate.SafetyRatings?.Select(sr => new Dictionary<string, object?>
```

Comment:
I was thinking here about create class like MetadataSafetyRatings but It will need casting when someone want to need access SafetyRatings metadata. What do you think @stephentoub ? I don't know do you allow to leak implementation details so used Dictionary.